.cloudtest-single-cluster.yaml

```diff
@@ -0,0 +1,46 @@
+---
+version: 1.0
+root: "./.tests/cloud_test/single-cluster/"
+timeout: 7200  # 2 hour total total timeout
+shuffle-enabled: false
+statistics:
+  enabled: true
+  interval: 30  # 30 seconds for statistics
+providers:
+  - name: "kind"    # A name of our provider
+    kind: "shell"   # A type of provider, shell indicate we need to specify all operations with shell commands.
+    instances: 1    # We need to have 1 instance of cloud.
+    node-count: 3   # With 3 nodes, CloudTest will validate if cloud has 3 nodes alive.
+    retry: 1        # A retry count if cloud is failed to start tool will try again.
+    enabled: true   # Indicate Provider is active with config by default, if set to false, values could be overriden using command line arguments.
+    timeout: 300    # 5 minutes to start cluster
+    stop-delay: 10
+    env:
+      - KIND_CLUSTER_NAME=cloudtest-kind-$(rands10)   # Generate a uniq cluster name
```

Comment:
How does the injection of the `$(rands10)` actually work?  Does cloudtest process the yaml file content in a way that a bash shell would exec that?

Reply:
Oh I see that a makefile target is wrapping usage of the env var with `$()` exec... interesting

Reply:
Cloudtest does do extra processing. 
https://github.com/networkservicemesh/cloudtest/blob/master/docs/README.md#environment-variables-processing

you get environment variables and a handful of other useful commands, like rands10. I'm not sure if this is executed by bash or not, they may have just borrowed the syntax.